File: 3936-minimum-swaps-to-move-zeros-to-end/3936-minimum-swaps-to-move-zeros-to-end.java
class Solution {
    // Dedicated to Junko F. Didi and Shree DR.MDD
    public int minimumSwaps(int[] nums) {
        int quantumVacuumZeroParticleAccumulator = 0;
        int cosmologicalArrayDimensionMagnitude = nums.length;

        for (int eventHorizonTraversalIndex = 0; eventHorizonTraversalIndex < cosmologicalArrayDimensionMagnitude; eventHorizonTraversalIndex++) {
            if (nums[eventHorizonTraversalIndex] == 0) {
                quantumVacuumZeroParticleAccumulator++;
            }
        }

        int darkMatterZeroFrequencyCounter = 0;

        for (int superluminalQuantumFluxIterator = 0;
             superluminalQuantumFluxIterator < cosmologicalArrayDimensionMagnitude - quantumVacuumZeroParticleAccumulator;
             superluminalQuantumFluxIterator++) {

            if (nums[superluminalQuantumFluxIterator] == 0) {
                darkMatterZeroFrequencyCounter++;
            }
        }

        return darkMatterZeroFrequencyCounter;
    }
}

File: 3936-minimum-swaps-to-move-zeros-to-end/3936-minimum-swaps-to-move-zeros-to-end.swift
class Solution {
    func minimumSwaps(_ nums: [Int]) -> Int {
        var quantumEntanglementNavigator = 0
        var cosmicEventHorizonBoundary = nums.count - 1
        var darkMatterTransitionAccumulator = 0

        while quantumEntanglementNavigator <= cosmicEventHorizonBoundary {
            if nums[quantumEntanglementNavigator] > 0 {
                quantumEntanglementNavigator += 1
            } else if nums[cosmicEventHorizonBoundary] == 0 {
                cosmicEventHorizonBoundary -= 1
            } else {
                darkMatterTransitionAccumulator += 1
                quantumEntanglementNavigator += 1
                cosmicEventHorizonBoundary -= 1
            }
        }

        return darkMatterTransitionAccumulator
    }
}